### PR TITLE
Fix typescript jest issue in `forgotPassword.test.ts`

### PR DESCRIPTION
**Pull request by Roesh**
Closes: #2089

### What are the changes and their implications?
Modified ...jest.requireActual("blitz")!,to ...jest.requireActual<object>("blitz")!, to prevent typescript compiler from throwing error.

Using the <object> for now, I tried to get the type of that object using a console log but it said "object" as well, but if there's a better type to assign here sometime in the future, we can swap object out with that instead.
console.log(typeof jest.requireActual("@blitzjs/core/server")!)
Console
    console.log
      object
      at app/auth/mutations/forgotPassword.test.ts:12:11

In addition, I also updated the auth examples tests files to be the same as the templates test files

### Checklist

- [ ] Changes covered by tests (tests added if needed)
- [ ] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes

### CODE REVIEW

File: examples/auth/app/auth/mutations/forgotPassword.test.ts [hunk 1]
```diff
@@ -8,15 +8,15 @@ beforeEach(async () => {
 })
 
 const generatedToken = "plain-token"
-jest.mock("@blitzjs/core/server", () => ({
-  ...jest.requireActual("@blitzjs/core/server")!,
+jest.mock("blitz", () => ({
+  ...jest.requireActual<object>("blitz")!,
```
Comment (flybayer) [inline examples/auth/app/auth/mutations/forgotPassword.test.ts:12]: Thanks @Roesh!

You have to actually revert this change as it breaks example tests here in the monorepo. Maybe a way to fix it, but needs to be `@blitzjs/core/server` for now

Comment (Roesh) [reply examples/auth/app/auth/mutations/forgotPassword.test.ts]: Ah, got it. Let me revert that

File: examples/auth/app/pages/index.test.tsx [hunk 2]
```diff
@@ -1,27 +1,25 @@
...
+  })
 
-  const {getByText} = render(<Home />)
-  const element = getByText(/powered by blitz/i)
-  // @ts-ignore
-  expect(element).toBeInTheDocument()
+  const { getByText } = render(<Home />)
+  const linkElement = getByText(/Documentation/i)
+  expect(linkElement).toBeInTheDocument()
 })
```
Comment (flybayer) [inline examples/auth/app/pages/index.test.tsx:25]: So some change in this file is still causing tests to fail.

Not sure if `jest.mock` needs to be before the import?

Comment (Roesh) [reply examples/auth/app/pages/index.test.tsx:25]: Ah I see. Should the test syncing be a separate PR then? I can commit just the single change to satisfy typescript and we can update the auth tests later? 
Also I'm actually not sure how to trigger the CI/CD tests, is there a way to retrigger them and check if the changes I made are OK in the monorepo?

Comment (flybayer) [reply examples/auth/app/pages/index.test.tsx:25]: it's up to you. If we get tests passing, then fine to keep the change here.

CI auto runs on each `git push`, but you can run these failing tests locally by running `yarn jest` inside `examples/auth`

Comment (Roesh) [reply examples/auth/app/pages/index.test.tsx]: Ah, I see. I reverted back to the original tests, with the only change being the <object> before the tests. I tried to troubleshoot the index.test.tsx file but I'm actually ran into issues with the forgotpassword mutation as well, so I reverted that back to its original version too. I made sure to run yarn jest this time and all tests passed.

Comment (Roesh) [reply examples/auth/app/pages/index.test.tsx]: Ok, that was on me, I forgot to push the changes I'd made after resolving a merge conflict, I reran the yarn jest command and eslint --fix and lint, and those passed and waiting for the results of the CI tests

**Comment by flybayer** [comment]
Thank you!!
